pilot/pkg/xds/filters/filters.go

```diff
@@ -163,9 +163,12 @@ var (
 	// These ALPNs are injected in the client side by the ALPN filter.
 	// "istio" is added for each upstream protocol in order to make it
 	// backward compatible. e.g., 1.4 proxy -> 1.3 proxy.
-	mtlsHTTP10ALPN = []string{"istio-http/1.0", "istio"}
-	mtlsHTTP11ALPN = []string{"istio-http/1.1", "istio"}
-	mtlsHTTP2ALPN  = []string{"istio-h2", "istio"}
+	// Non istio-* variants are added to ensure that traffic sent out of the mesh has a valid ALPN;
+	// ideally this would not be added, but because the override filter is in the HCM, rather than cluster,
```

Comment:
This PR https://github.com/envoyproxy/envoy/pull/13922 adds ALPN   options to cluster.  I have n't through your PR fully,  but FYI so that you are aware and think about any changes needed/improvements given that it is supported at cluster now.